src/pages/variantHistory/index.tsx

```diff
@@ -38,32 +40,56 @@ const VariantHistoryContents: React.VFC = () => {
     variantName: string;
   }>();
   const { sendEvent } = useProjectHealthAnalytics({ page: "Variant history" });
+  const { ingestNewCommits } = useHistoryTable();
+  const dispatchToast = useToastContext();
   usePageTitle(`Variant History | ${projectIdentifier} | ${variantName}`);
-  const [nextPageOrderNumber, setNextPageOrderNumber] = useState(null);
   useJumpToCommit();
   useTestFilters();
   const { badges, handleOnRemove, handleClearAll } = useFilterBadgeQueryParams(
     constants.queryParamsToDisplay
   );
-  const { data, error } = useQuery<
+  const { data, loading, refetch } = useQuery<
```

Comment:
I couldn't get this to work. afaict `fetchMore` requires the apollo cache to function. You trigger a fetch and then ask apollo to merge it with the previous cache entry by defining a merge function in the `inMemoryCache` . Since we aren't using the apollo cache for this component it doesn't seem to have an effect. I'm open to any suggestions on how to get it to work.

Reply:
Interesting, if I'm understanding correctly it seems like the [commitCache](https://github.com/evergreen-ci/spruce/blob/58d495369cb3e091063d79fc1ba3ba9b6d8749bb/src/components/HistoryTable/historyTableContextReducer.ts#L27) is used with [objectifyCommits](https://github.com/evergreen-ci/spruce/blob/58d495369cb3e091063d79fc1ba3ba9b6d8749bb/src/components/HistoryTable/historyTableContextReducer.ts#L208-L223) for merging new data with the old as it comes in. We could probably just offload this to Apollo altogether though and add that logic to a merge function. Not necessary for this ticket but do you think it'd be worth doing generally?

Reply:
Just interested in your thoughts here, not a blocker at all.

Reply:
Thats a good point tbh taking a look at the design of this component `commitCache` is superfluous and just duplicates the data in `processedCommits` Originally it was intended to be a mechanism for deduplicating the data that is loaded but it can easily be accomplished some other way. I opened https://jira.mongodb.org/browse/EVG-20212 to look into refactoring this.